server/rss/queries.js

```diff
@@ -1,122 +1,257 @@
+						as: 'exports',
+					},
 				],
-				separate: true,
+			},
+			{
+				model: PubEdge,
+				as: 'inboundEdges',
+			},
+			{
+				model: PubEdge,
+				as: 'outboundEdges',
+			},
+			{
+				model: CollectionPub,
+				as: 'collectionPubs',
 				include: [
 					{
-						model: PubAttribution,
-						as: 'attributions',
-						required: false,
+						required: true,
+						where: {
+							isPublic: true,
+						},
+						model: Collection,
+						as: 'collection',
 						include: [
-							includeUserModel({
-								as: 'user',
-								required: false,
-							}),
+							{
+								model: CollectionAttribution,
+								as: 'attributions',
+								include: [includeUserModel({ as: 'user' })],
+							},
 						],
 					},
-					{
-						model: Release,
-						as: 'releases',
-					},
 				],
 			},
 		],
-	}).then((communityData) => {
-		if (!communityData) {
-			throw new Error('Community Not Found');
-		}
-		const communityUrl = communityData.domain
-			? `https://${communityData.domain}`
-			: `https://${communityData.subdomain}.pubpub.org`;
-		const feed = new RSS({
-			title: communityData.title,
-			description: communityData.description,
-			feed_url: `${communityUrl}/rss.xml`,
-			site_url: communityUrl,
-			image_url: communityData.favicon,
-			webMaster: 'hello@pubpub.org',
-			language: 'en',
-			pubDate: new Date(),
-			ttl: '60',
-		});
-		communityData.pubs
-			.map((pub) => {
-				const pubJSON = pub.toJSON();
-				const publishedDate = getPubPublishedDate(pub);
-				if (publishedDate) {
-					pubJSON.listedDate = publishedDate;
-				}
-				return pubJSON;
-			})
-			.filter((pub) => {
-				return pub.listedDate;
-			})
-			.sort((foo, bar) => {
-				if (foo.listedDate < bar.listedDate) {
-					return 1;
-				}
-				if (foo.listedDate > bar.listedDate) {
-					return -1;
-				}
-				return 0;
-			})
-			.forEach((pub) => {
-				const authors = pub.attributions
-					.map((attribution) => {
-						if (attribution.user) {
-							return attribution;
-						}
-						return ensureUserForAttribution(attribution);
-					})
-					.filter((item) => {
-						return item.isAuthor;
-					})
-					.map((item) => {
-						return item.user.fullName;
-					})
-					.join(', ');
+	});
+	// pubIds are already in publication-date order, so sort results based on that.
+	return pubs.sort((a, b) => pubIds.indexOf(a.id) - pubIds.indexOf(b.id));
+};
 
-				feed.item({
-					title: pub.title,
-					description: pub.description,
-					url: `${communityUrl}/pub/${pub.slug}`,
-					guid: pub.id,
-					author: authors,
-					date: pub.listedDate,
-					enclosure: {
-						url: pub.avatar,
-					},
-				});
-			});
+const createEnclosure = (url) => {
+	return {
+		enclosure: {
+			_attr: {
+				url: url,
+			},
+		},
+	};
+};
 
-		return feed.xml();
+export const getFeedItemForPub = (pubData, communityData) => {
+	const { title, description, collectionPubs, id } = pubData;
+	const formattedDownload = getFormattedDownloadUrl(pubData);
```

Comment:
Nit but I think `formattedDownloadUrl`  might be a more descriptive variable name.

Reply:
oh yeah totally